providers/base/bin/camera_test.py

```diff
@@ -711,14 +955,17 @@ def get_video_devices():
         gi.require_version("GLib", "2.0")
         from gi.repository import GLib
 
-        gi.require_version("Clutter", "1.0")
-        from gi.repository import Clutter
+        Gst.init(None)
+        camera.Gst = Gst
+        camera.GLib = GLib
+
+        # Import Gtk only for the test cases that will need it
+        if args["test"] in ["video", "image"] and not args["quiet"]:
 
-        gi.require_version("Gtk", "3.0")
-        from gi.repository import Gtk
+            gi.require_version("Gtk", "3.0")
+            from gi.repository import Gtk
 
-        Gst.init(None)
-        Clutter.init()
-        Gtk.init([])
-    camera = CameraTest(args)
-    sys.exit(getattr(camera, args.test)())
+            Gtk.init([])
+            camera.Gtk = Gtk
```

Comment:
ditto, we can have an `init_gtk` function that is called when needed and if gtk is not inited, it gets imported and inited. Any reason why this was done thisway?

Reply:
Done